bpf/lib/lb.h

```diff
@@ -574,17 +573,29 @@ struct lb6_service *lb6_lookup_service(struct lb6_key *key,
 	svc = map_lookup_elem(&LB6_SERVICES_MAP_V2, key);
 	if (svc) {
 		if (!scope_switch || !lb6_svc_is_local_scope(svc))
-			/* Packets for L7 LB are redirected even when there are no backends. */
-			return (svc->count || lb6_svc_is_l7loadbalancer(svc)) ? svc : NULL;
+			return svc;
 		key->scope = LB_LOOKUP_SCOPE_INT;
 		svc = map_lookup_elem(&LB6_SERVICES_MAP_V2, key);
-		if (svc && (svc->count || lb6_svc_is_l7loadbalancer(svc)))
-			return svc;
+		return svc;
 	}
 
 	return NULL;
 }
 
+static __always_inline
+struct lb6_service *lb6_lookup_service(struct lb6_key *key,
```

Comment:
Can you add a note to the helpers highlighting what the difference is?

Reply:
Done, thanks!